newIDE/electron-app/app/main.js

```diff
@@ -218,6 +220,9 @@ app.on('ready', function() {
     mainWindow.webContents.send('yarn-changes-saved', newFilePath);
   });
 
+  // Discord Rich Presence event:
+  ipcMain.on('update-discord-rp', (event, config) => discordClient.updatePresence(config));
+
```

Comment:
nitpicking: could you put this in a "DiscordRichPresence.js" file? Both the require, the error handling, and the ipcMain.on (that you setup using an export function `setupDiscordRichPresence`). Just to make the main.js as agnostic as possible toward these extra features.